Server/DeviceManagement/DeviceManagement.Domain/UserAggregate/IUserDataSource.cs

```diff
@@ -7,6 +7,7 @@ namespace ModularHouse.Server.DeviceManagement.Domain.UserAggregate;
 
 public interface IUserDataSource
 {
-    Task<IReadOnlyList<User>> GetAllAsync(CancellationToken cancellationToken);
-    Task<User> GetByIdAsync(Guid id, CancellationToken cancellationToken);
+    Task<IReadOnlyList<User>> GetAllAsync(CancellationToken cancellationToken = default);
+    Task<User> GetByIdAsync(Guid id, CancellationToken cancellationToken = default);
+    Task<IReadOnlyList<Guid>> GetAreaIdsAsync(Guid id, CancellationToken cancellationToken = default);
```

Comment:
It's User's DataSource, so it should not contain logic out of it's boundaries.
In method GetAreaIdsAsync you return Ids of Areas, but inside of UserDataSource.
If you really need the behavior/logic like this (try to avoid this, CRUD are simple operations), you can use `AreaDataSource.GetAllAsync(AreaFilter filter).Select(x => x.Id)` by passing the data like `filter.UserId = ...`. Or you can add new method like `AreaDataSource.GetAllIdsAsync(AreaFilter filter)` which will return only list of Ids